edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -0,0 +1,805 @@
+                });
+            var deploymentsUpdated = new List<V1Deployment>();
+            var newDeployments = new List<V1Deployment>();
+            List<V1Deployment> currentDeploymentsList = currentDeployments.Items.ToList();
+            desiredDeployments.ForEach(
+                d =>
+                {
+                    if (currentDeploymentsFromAnnotations.ContainsKey(d.Metadata.Name))
+                    {
+                        V1Deployment current = currentDeploymentsList.Find(i => string.Equals(i.Metadata.Name, d.Metadata.Name));
+                        string currentFromAnnotation = currentDeploymentsFromAnnotations[d.Metadata.Name];
+                        string creationString = JsonConvert.SerializeObject(d);
+
+                        // If configuration matches, or this is edgeAgent deployment and the images match,
+                        // no need to do update deployment
+                        if (string.Equals(currentFromAnnotation, creationString) ||
+                            (string.Equals(d.Metadata.Name, this.DeploymentName(CoreConstants.EdgeAgentModuleName)) && V1DeploymentEx.ImageEquals(current, d)))
+                        {
+                            return;
+                        }
+
+                        d.Metadata.ResourceVersion = current.Metadata.ResourceVersion;
+                        if (d.Metadata.Annotations == null)
+                        {
+                            var annotations = new Dictionary<string, string>
+                            {
+                                [Constants.CreationString] = creationString
+                            };
+                            d.Metadata.Annotations = annotations;
+                        }
+                        else
+                        {
+                            d.Metadata.Annotations[Constants.CreationString] = creationString;
+                        }
+
+                        deploymentsUpdated.Add(d);
+                        this.logger.UpdateDeployment(d.Metadata.Name);
+                    }
+                    else
+                    {
+                        string creationString = JsonConvert.SerializeObject(d);
+                        var annotations = new Dictionary<string, string>
+                        {
+                            [Constants.CreationString] = creationString
+                        };
+                        d.Metadata.Annotations = annotations;
+                        newDeployments.Add(d);
+                        this.logger.CreateDeployment(d.Metadata.Name);
+                    }
+                });
+
+            // Remove the old
+            IEnumerable<Task<V1Status>> removeServiceTasks = servicesRemoved.Select(
+                i =>
+                {
+                    this.logger.DeletingService(i);
+                    return this.client.DeleteNamespacedServiceAsync(i.Metadata.Name, KubeUtils.K8sNamespace, new V1DeleteOptions());
```

Comment:
>KubeUtils.K8sNamespace [](start = 85, length = 22)

You should remove the KubeUtils.K8sNamespace property to catch all the places where we need to replace it with the namespace passed in.